src/Components/Common/WizardContext/initialState.js

```diff
@@ -2,8 +2,11 @@ const initialWizardContext = {
   chosenSource: undefined,
   chosenNumOfInstances: 1,
   chosenInstanceType: undefined,
+  uploadedKey: false,
   sshPublicName: '',
   sshPublicKey: '',
+  chosenSshKeyId: undefined,
+  chosenSshKeyName: null,
```

Comment:
why not use `sshPublicName`?

Reply:
the same reason as above, they could collide, although here we could probably fix that by nullifying the vars on switching, but then you'd loose what you've filled out when you:
1. go to upload a key, you fill out everyting
2. you wanna check you do not have some key that you remembered you might have
3. you switch to a choose public key
4. you go throught them, select some by accident
5. you go back to uploading

IMHO in this scenario the public key should not be gone. I'll write tests for that.

Reply:
also, I think that `sshPublicName` should be renamed at some point, but that is irrelevant to this discussion :)

Reply:
as you mentioned, this scenario can be fixed by just resetting the state on switching. I'd prefer fewer states, especially when the state is shareable among the whole wizard though I won't block this PR on this, we can improve this later on (i.e better naming `sshPublicName` ->  `uploadedSSHKeyName`)

Reply:
I've added test for the scenario I've mentioned above, I believe simple reuse of the same state would break the test now.
Take a look at the test if you agree please :)